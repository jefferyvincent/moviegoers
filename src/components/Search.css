.app-search {
    background-color: #3b3b3b;
    border-radius: 3px;
    float: left;
    height: 54px;
    margin: 0;
    position: relative;
    width: 100%;
    z-index: 9999999;
}

.app-search input {
    outline: none;
    background-color: transparent;
    border: none;
    color: #dcdcdc;
    font-size: 20px;
    height: 100%;
    left: 0;
    opacity: 1;
    padding: 0px 46px 15px 15px;
    position: absolute;
    top: 10px;
    width: 100%;
    font-weight: 200;
}

.search-mag {
    position: absolute;
    top: 7px;
    right: 10px;
}

.search-results {
    background-color: rgba(34,34,34,.97);
    font-size: 16px;
    height: auto;
    max-height: 300px;
    overflow-x: auto;
    left: 0;
    line-height: 30px;
    list-style: none;
    margin: 0;
    padding: 53px 20px 20px 25px;
    position: absolute;
    width: 100%;
    top: 54px;
}

.search-results li {
    clear: both;
    overflow: hidden;
    margin: 0 0 4px;
    float: left;
    width: 100%;
    text-align: left;
}

.search-results li a {
    overflow: hidden;
    display: block;
    padding: 4px;
    line-height: 31px;
    font-weight: 400;
    color: white;
    text-decoration: none;
}
.search-results li a img {
    max-width: 100px;
}

.img-wrap {
    float: left;
    margin: 0 16px 0 0;
    width: 60px;
    height: 60px;
    border-radius: 100px;
    position: relative;
    overflow: hidden
}



@media only screen and (max-width: 904px) {
    .app-search {
        height: 40px;
    }
}