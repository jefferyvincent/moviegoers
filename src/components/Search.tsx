import React, {useState, useEffect} from 'react';
import axios from 'axios';
import { Link } from "react-router-dom";

import './Search.css';
import mag from '../img/mag.svg';

const Search: React.FC = () => {
    const [data, setData] = useState();
    const [query, setQuery] = useState();
    const [hidden, setHidden] = useState('hidden')

    useEffect(() => {
        if (query !== '') {
            const fetchData = async () => {
                const result = await axios(
                `https://api.themoviedb.org/3/search/movie?query=${query}&api_key=438581175542f3ba5208b33d46f87151`,
                );
                setData(result.data);
            };
            fetchData();
        }
    }, [query]);

    const renderData = () => {
        if (data !== undefined) {
            if (data.hasOwnProperty('results')) {
                return (
                    <ul className={"Search-results "+ hidden}>
                        {data.results.map((item:any) => (
                            <li key={item.id}>
                                <Link to={"/movie/"+item.id}>
                                    <div className="img-wrap">
                                        <img src={"https://image.tmdb.org/t/p/w500/"+item.poster_path} alt={item.title} />
                                    </div>
                                    {item.title}
                                </Link>
                            </li>
                        ))}
                    </ul>
                );
            }
        } else {
            return null;
        }
    }

    const showResults = ():void => {
        setHidden('');
        document.addEventListener('mousedown', hideResults, false);
        document.onkeydown = function(evt) {
            evt = evt || window.event;
            if (evt.keyCode === 27) { // esc
                hideResults();
            }
        };
    }

    const hideResults = ():void => {
        setTimeout(()=>{
            setHidden('hidden');
            document.removeEventListener('mousedown', hideResults, false);
        }, 1000)
    }

    return (
        <div className="App-search">
            <input autoComplete="off" onChange={(evt)=>{ showResults(); setQuery(evt.target.value); } } autoCorrect="off" spellCheck={false} placeholder="search it. watch it"></input>
            <img className="Search-mag" src={mag} alt="Search" />
            {renderData()}
        </div>
    );
}

export default Search;