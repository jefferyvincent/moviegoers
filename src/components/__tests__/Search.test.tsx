import React from 'react';
import { shallow } from 'enzyme';
import {render } from 'react-testing-library';
import Search from '../Search';



describe('<Search/>', () => {
  let wrapper:any;
  beforeEach(() => { wrapper = shallow(<Search />); });

  it("renders", () => {
    const { asFragment } = render(<Search />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders an input area', () => {
    expect(wrapper.find('input').length).toEqual(1);
  });

});