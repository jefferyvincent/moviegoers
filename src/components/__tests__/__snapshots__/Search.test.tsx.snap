// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search/> renders 1`] = `
<DocumentFragment>
  <div
    class="app-search"
  >
    <input
      autocomplete="off"
      autocorrect="off"
      placeholder="search it."
      spellcheck="false"
    />
    <img
      alt="Search"
      class="search-mag"
      src="mag.svg"
    />
  </div>
</DocumentFragment>
`;
